pkg/user/casbin/rbac.go

```diff
@@ -487,3 +493,191 @@ func MatchKeyByPart(key1 string, key2 string) bool {
 	}
 	return true
 }
+
+func (e *EnforcerImpl) BatchEnforceForSubject(subject string, rvalsArr [][]interface{}) []bool {
+	var resultArr []bool
+
+	functions := make(map[string]govaluate.ExpressionFunction)
+	enforcedModel := e.Enforcer.GetModel()
+	fm := model.LoadFunctionMap()
+	for key, function := range fm {
+		functions[key] = function
+	}
+	functions["matchKeyByPart"] = MatchKeyByPartFunc
```

Comment:
golang provides mechanism to add custom function, i think vikram has done it at some other place